### PR TITLE
VSP-479 [iOS] Forgot password is not verifying if it is an email.

### PR DESCRIPTION
**Pull request by luciancerbu-vsp**
- Fixed forgot password issue, where there was no validation of email inserted by the user.

- The same issue was fixed in Account details screen.

### CODE REVIEW

File: Permanent/ViewControllers/Authentication/LoginViewController.swift [hunk 1]
```diff
@@ -107,7 +107,18 @@ class LoginViewController: BaseViewController<AuthViewModel> {
             showAlert(title: .success,
                       message: String(format: .emailSent, email!))
         case .error(let message):
-            showAlert(title: .error, message: message)
+            DispatchQueue.main.async {
```
Comment (v-rusu) [inline Permanent/ViewControllers/Authentication/LoginViewController.swift:110]: Please double check if this async dispatch is really needed. I suppose not

File: Permanent/ViewControllers/Authentication/LoginViewController.swift [hunk 2]
```diff
@@ -107,7 +107,18 @@ class LoginViewController: BaseViewController<AuthViewModel> {
...
+                let alert = UIAlertController(title: .error, message: message, preferredStyle: .alert)
+                
+                alert.addAction(UIAlertAction(title: .cancel, style: .cancel, handler: nil))
+                alert.addAction(UIAlertAction(title: .retry, style: .default, handler: {_ in
+                    self.dismiss(animated: true, completion: {
```
Comment (v-rusu) [inline Permanent/ViewControllers/Authentication/LoginViewController.swift:115]: Theoretically the action is executed after dismissal of the alert. can you please double check?

File: Permanent/ViewControllers/Authentication/LoginViewController.swift (hunk 1 above)
Comment (luciancerbu-vsp) [reply Permanent/ViewControllers/Authentication/LoginViewController.swift:110]: Checked and it works great without dispatch.

File: Permanent/ViewControllers/Authentication/LoginViewController.swift (hunk 2 above)
Comment (luciancerbu-vsp) [reply Permanent/ViewControllers/Authentication/LoginViewController.swift:115]: Checked and there is no need for dismiss.

**Review by cgeorgiuvsp — APPROVED** [review]

**Pull request merged**